export/pdf-component.js

```diff
@@ -47,68 +49,97 @@ class PrintToPdf extends HTMLElement {
       composed: true
     });
     this.shadowRoot.dispatchEvent(event);
-  };
+  }
 
   _printPdf() {
     window.print();
-  };
+  }
 
   _scrollTop() {
     const element = this.shadowRoot.querySelector('#element-to-print');
     element.scrollTop = 0;
-  };
-
-  _downloadPdf(event) {
-    const elementToPrint = this.shadowRoot.querySelector('#element-to-print');
+  }
 
-    // Safari doesn't support huge canvas sizes, we lower the resolution for
-    // Apple devices to make sure the PDF fits on the canvas.
-    const isApple = navigator.vendor.match(/apple/i);
+  _dispatchProgress(message) {
+    const progressEvent = new CustomEvent('progress', {
+      bubbles: true,
+      composed: true,
+      detail: {message}
+    });
+    this.dispatchEvent(progressEvent);
+  }
 
-    // Firefox does not support a canvas size as big as chrome
-    const isFirefox = navigator.userAgent.match(/firefox/i)
+  // Copypaste from https://stackoverflow.com/questions/60234692/how-to-merge-two-pdf-files-using-jspdf
+  async mergePdfs(pdfsToMerges) {
+    const mergedPdf = await PDFDocument.create();
+    const actions = pdfsToMerges.map(async pdfBuffer => {
+      const pdf = await PDFDocument.load(pdfBuffer);
+      const copiedPages = await mergedPdf.copyPages(pdf, pdf.getPageIndices());
+      copiedPages.forEach((page) => {
+        mergedPdf.addPage(page);
+      });
+    });
+    await Promise.all(actions);
+    return await mergedPdf.save();
+  }
 
-    // Do not abuse, higher values are glitchy and images may disappear or the
-    // file may fail to render altogether
-    let scale = 4;
-    if (isFirefox) {
-      scale = 3;
-    }
-    if (isApple) {
-      scale = 1;
-    }
+  _downloadPdf(event) {
+    const elementToPrint = this.shadowRoot.querySelector('#element-to-print');
 
     if (!elementToPrint) {
       console.warn('The web component has not rendered yet, retrying in 100ms');
-      setTimeout( () => this._downloadPdf(event), 100);
-      return
+      setTimeout(() => this._downloadPdf(event), 100);
+      return;
     }
     try {
       const options = {
         image: {
-            type: 'jpeg',
-            quality: 0.75
-          },
-          html2canvas: {
-            scrollX: 0,
-            scrollY: 0,
-            scale: scale
-          }
-      }
-      html2pdf()
-        .set(options)
-        .from(elementToPrint)
-        .toPdf()
-        .save(event.detail?.fileName || 'file.pdf')
-        .then(() => {
-          const downloadedEvent = new Event('downloaded');
-          this.dispatchEvent(downloadedEvent);
+          type: 'jpeg',
+          quality: 0.75
+        },
+        html2canvas: {
+          scrollX: 0,
+          scrollY: 0,
+          scale: 4
+        }
+      };
+
+      const pages = [...elementToPrint.children[0].children];
+
+      const pagesPdf = [];
+
+      pages.forEach((page, index) => {
+        const pdf = html2pdf().set(options).from(page).toPdf().get('pdf');
```

Comment:
It is no possible to use one library for both cases (generate a pdf page and also the merge of the pages)? So we dont need two libraries for this?

Reply:
As far as I know, no. jspdf/html2pdf is used to render the HTML into a canvas and then a pdf. pdf-lib is a pdf manipulation library, and it does not support generating PDF from html. Their scope is different.